v2/handler.js

```diff
@@ -966,3 +971,33 @@ function InRequestOptions(
     self.hostPort = hostPort;
     self.tracer = tracer;
 }
+
+function validateHostPort(hostPort) {
+    var parts = hostPort.split(':');
+    if (parts.length !== 2) {
+        return false;
+    }
+
+    if (parts[0] === '127.0.0.1') {
+        return false;
+    }
+
+    if (parts[0] === '0.0.0.0') {
```

Comment:
This is the host port as passed in the header of an init request. The assertion error we were seeing was in [peer.js:566](https://github.com/uber/tchannel-node/blob/master/peer.js#L569): `assert(parts.length === 2, 'invalid destination ' + self.hostPort);`, and it's the host port used when we instantiate a peer object, which comes from the header of an init request. This happens far before the advertise path.

It's true that we allow a hostPort of `0.0.0.0:0` for ephemeral clients ([connection.js:519](https://github.com/uber/tchannel-node/blob/master/connection.js#L519)), so this particular conditional should be removed; but I do believe that this is the correct place for this check.